src/Service/QuizInstanceService.php

```diff
@@ -128,13 +129,28 @@ public function getResultsNumber(Criteria $criteria): int
         return $quizInstances->getCount();
     }
 
+    /**
+     * Marks the completion of a quiz in the database.
+     *
+     * @param int $quizInstanceId
+     * @param bool $isComplete
+     */
+    public function markQuizComplete(int $quizInstanceId, bool $isComplete): void
```

Comment:
this naming is an issue. The method says "mark quiz complete" but if the `$isComplete` argument is `false`, then it actually marks it as incomplete. Maybe change it to `markQuiz`